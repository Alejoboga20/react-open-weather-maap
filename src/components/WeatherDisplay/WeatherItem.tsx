import '../../styles/components/WeatherItem.css';

export interface WeatherItemProps {
  temp: number;
}

export const WeatherItem = ({ temp }: WeatherItemProps) => (
  <div className='weatherItem_container'>
    <p>{temp} Â°</p>
  </div>
);
