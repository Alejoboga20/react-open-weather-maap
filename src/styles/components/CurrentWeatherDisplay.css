.weather_container > p {
  font-size: larger;
}
