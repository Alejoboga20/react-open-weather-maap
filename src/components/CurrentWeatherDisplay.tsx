import '../styles/components/WeatherDisplay.css';

interface CurrentWeatherDisplayProps {
  temperature?: number;
}

export const CurrentWeatherDisplay = ({
  temperature,
}: CurrentWeatherDisplayProps) => {
  return (
    <div className='container'>
      {temperature && <p>{`It is ${temperature} degress in the city`}</p>}
    </div>
  );
};
