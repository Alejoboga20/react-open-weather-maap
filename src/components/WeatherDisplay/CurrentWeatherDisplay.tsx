import '../../styles/components/CurrentWeatherDisplay.css';
import { Label } from '../common/Label';

interface CurrentWeatherDisplayProps {
  temperature?: number;
}

export const CurrentWeatherDisplay = ({
  temperature,
}: CurrentWeatherDisplayProps) => {
  return (
    <div className='weather_container'>
      {temperature ? (
        <Label labelContent={`It is ${temperature} degress in the city`} />
      ) : (
        <Label labelContent='Please Ensure your location service is working' />
      )}
    </div>
  );
};
