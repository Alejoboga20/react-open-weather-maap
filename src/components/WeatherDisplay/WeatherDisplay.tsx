import { Daily } from '../../interfaces/WeatherResponse';
import { Label } from '../common/Label';
import { WeatherItem } from './WeatherItem';
import '../../styles/components/WeatherDisplay.css';

export interface WeatherDisplayProps {
  dailyTemp?: Daily[];
}

export const WeatherDisplay = ({ dailyTemp }: WeatherDisplayProps) => {
  return (
    <div className='weatherDisplay_container'>
      {dailyTemp && (
        <Label labelContent='Temperature for the following days is: ' />
      )}
      {dailyTemp?.map((daily) => (
        <WeatherItem key={daily.dt} temp={daily.temp.day} />
      ))}
    </div>
  );
};
