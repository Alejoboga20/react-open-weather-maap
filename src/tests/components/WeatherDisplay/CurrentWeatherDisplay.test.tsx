import { shallow } from 'enzyme';
import { CurrentWeatherDisplay } from '../../../components/WeatherDisplay/CurrentWeatherDisplay';

const labelContent = 'It is 10 degress in the city';
const labelError = 'Please Ensure your location service is working';

describe('CurrentWeatherDisplay Tests', () => {
  const wrapper = shallow(<CurrentWeatherDisplay temperature={10} />);

  test('should match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should show label with temperature', () => {
    expect(wrapper.find('Label').prop('labelContent')).toBe(labelContent);
  });

  test('should show label with error', () => {
    const wrapper = shallow(<CurrentWeatherDisplay />);
    expect(wrapper.find('Label').prop('labelContent')).toBe(labelError);
  });
});
