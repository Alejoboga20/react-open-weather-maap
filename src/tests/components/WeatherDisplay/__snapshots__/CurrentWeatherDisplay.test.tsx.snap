// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeatherDisplay Tests should match Snapshot 1`] = `
<div
  className="weather_container"
>
  <Label
    labelContent="It is 10 degress in the city"
  />
</div>
`;
